kubernetes/modules/materialize-instance/main.tf

```diff
@@ -103,7 +103,7 @@ resource "kubernetes_secret" "materialize_backend" {
       persist_backend_url  = var.persist_backend_url
       license_key          = var.license_key == null ? "" : var.license_key
     },
-    var.authenticator_kind == "Password" && var.external_login_password_mz_system != null ? {
+    contains(["Password", "Sasl"], var.authenticator_kind) && var.external_login_password_mz_system != null ? {
```

Comment:
Adding support for SASL auth as a follow up to: https://github.com/MaterializeInc/materialize-terraform-self-managed/pull/115